[14.0] [MIG] purchase_order_import

rconjour opened the pull request.

alexis-via committed:
Initial check-in of modules purchase_order_import, purchase_order_import_ubl and purchase_order_ubl

Move XSD files to base_ubl, to avoid duplication of UBL XSD files.

alexis-via committed:
Initial check-in of modules purchase_order_import, purchase_order_import_ubl and purchase_order_ubl

alexis-via committed:
Update README.rst

alexis-via committed:
Call api.multi get_ubl_filename method on a real object

alexis-via committed:
Update code following latest update of base_ubl

alexis-via committed:
Support context key no_embedded_ubl_xml

alexis-via committed:
Add support for UBL 2.0

alexis-via committed:
Handle lang in UBL XML file generation

alexis-via committed:
FIX Don't use _inherit = ['business.document.import']

because we want to have access to the code of the modules that inherit business.document.import

Commit:
[IMP] handle the case where the xml file is generated with mime type
'text/xml'

alexis-via committed:
Add support for extraction + matching on website

PEP8 fix

Commit:
8.0 Add support for partner bank matching on invoice update (#6)

Add support for partner bank matching on invoice update (before, it was only supported on invoice creation)

Commit:
8.0 Add support for partner bank matching on invoice update (#6)

Add support for partner bank matching on invoice update (before, it was only supported on invoice creation)

oca-transbot committed:
OCA Transbot updated translations from Transifex

alexis-via committed:
Port purchase_order_ubl to v10

Prepare v10 branch

Rename __openerp__.py to __manifest__.py and set installable to False

Also port all the modules that generate the XML documents: account_invoice_ubl, account_invoice_zugferd, purchase_order_ubl and sale_order_ubl

Rename account_invoice_zugferd to account_invoice_factur-x

Rename account_invoice_import_zugferd to account_invoice_import_factur-x
Add module to support py3o reporting engine:

Continue port of modules for v10.0, in particular sale_order_import_* module

Fix spelling mistake and other remarks on README by Tarteo

Disable get_pdf() in all tests because it doesn't work in Travis

[10.0] restore get_pdf() in tests (#31)

* sale_order_ubl + purchase_order_ubl: restore get_pdf() in tests using HttpCase

* Restore get_pdf() in tests of account_invoice_factur-x and account_invoice_ubl modules

* Update oca_dependencies.txt

alexis-via committed:
Prepare v10 branch

Rename __openerp__.py to __manifest__.py and set installable to False

alexis-via committed:
Continue port of modules for v10.0, in particular sale_order_import_* module

Fix spelling mistake and other remarks on README by Tarteo

alexis-via committed:
Port purchase_order_import* to v10.0

Add ubl invoice generation option in accounting config page

astirpe committed:
[11.0][MIG] purchase_order_ubl

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-travis committed:
[UPD] Update purchase_order_import.pot

alexis-via committed:
Fix import of sale order with price_source='pricelist' when pricelist has visible discounts

Code improvements in sale_order_import
Add unit tests in sale_order_import
Use display_name instead of name_get()[0][1]

alexis-via committed:
Check the VAT number of the destination partner, to make sure the business document is imported in the right company

oca-travis committed:
[UPD] Update purchase_order_import.pot

luc-demeyer committed:
[FIX]support custom reports

OCA-git-bot committed:
[UPD] README.rst

oca-travis committed:
[UPD] Update purchase_order_ubl.pot

oca-transbot committed:
Update translation files

Updated by Actualizar ficheiros PO com o novo POT (msgmerge) hook in Weblate.

astirpe committed:
[12.0][MIG] purchase_order_ubl

astirpe committed:
[REF] Split purchase_order_ubl and purchase_stock_ubl

OCA-git-bot committed:
[ADD] icon.png

gurneyalex committed:
[FIX] 10.0 bad use of play_onchange

the method play_onchange in server-tools/onchange_helper
only returns the changed keys in the dictionary it receives as first arguement.
The code in account_invoice_import, purchase_order_import and sale_order_import
was written as if all the keys were returned. Maybe the behavior of play_onchange
was updated in a recent version, but this nevertheless needs fixing in this module.

This should fix the unit tests which are red on this branch.

gurneyalex committed:
[FIX] remove method defined twice

gurneyalex committed:
[FIX] purchase_order_import: missing price_unit on line

oca-travis committed:
[UPD] Update purchase_order_import.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: edi-10.0/edi-10.0-purchase_order_import
Translate-URL: https://translation.odoo-community.org/projects/edi-10-0/edi-10-0-purchase_order_import/

astirpe committed:
[13.0][MIG] purchase_order_ubl

lmignon committed:
[ADD] Add support for UBL OrderResponse
The ORderResponce document allows a supplier to communicate the ack and the confirmation of an Order documnet. Depending of its content, the PO will be cancelled or confirmed. If the OrderResponse document contains some amended line, the import process will update the picking to reflect the changes

lmignon committed:
[IMP] purchase_order_import: log exception in case of parsing error

bealdav committed:
IMP purch_order: allow to using xml with no pdf and hook to customize xml

astirpe committed:
pre-commit

astirpe committed:
Code review

[UPD] Update purchase_order_ubl.pot

[UPD] README.rst

[ADD] icon.png

OCA-git-bot committed:
purchase_order_ubl 13.0.1.1.0

bealdav committed:
FIX purchase_order_import_*: version number after manual merge

oca-travis committed:
[UPD] Update purchase_order_import.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: edi-10.0/edi-10.0-purchase_order_import
Translate-URL: https://translation.odoo-community.org/projects/edi-10-0/edi-10-0-purchase_order_import/

simahawk committed:
Add base_edi

simahawk committed:
Add edi module

oca-travis committed:
[UPD] Update base_edi.pot

OCA-git-bot committed:
[UPD] README.rst

etobella committed:
[IMP] edi: define type, record, record create helper

Co-Authored-By: Simone Orsi <simahawk@gmail.com>

etobella committed:
[IMP] edi: backend add generate_output

Co-Authored-By: Simone Orsi <simahawk@gmail.com>

etobella committed:
[IMP] edi: Set Send function

Co-Authored-By: Simone Orsi <simahawk@gmail.com>

etobella committed:
[IMP] edi: Define a processing function for incoming documents

Co-Authored-By: Simone Orsi <simahawk@gmail.com>

simahawk committed:
edi: small fix/imp on models

simahawk committed:
edi: use freezegun for tests

simahawk committed:
edi: exchange type validate backend vs type

etobella committed:
[IMP] base_edi: Add Icon

simahawk committed:
edi bump 13.0.1.1.0

OCA-git-bot committed:
[UPD] README.rst

etobella committed:
edi bump 13.0.1.2.0

simahawk committed:
edi: lookup components by convention

simahawk committed:
edi: exchange record backend mandatory

simahawk committed:
edi: exchange type backend ondelete=set null

etobella committed:
[IMP] edi: Add mixins

oca-travis committed:
[UPD] Update edi.pot

oca-travis committed:
[UPD] Update edi.pot

OCA-git-bot committed:
edi 13.0.1.3.0

oca-travis committed:
[UPD] Update edi.pot

OCA-git-bot committed:
edi 13.0.1.4.0

simahawk committed:
edi.backend: unify component usage get

simahawk committed:
edi.exchange.record: add handy methods to set/get file

simahawk committed:
edi.backend: add cron to handle record state updates

simahawk committed:
Add edi_storage

simahawk committed:
edi: add specific events, improve record msg lookup

simahawk committed:
edi: pass backend to component ctx

simahawk committed:
edi.exchange.record: improve set/get file

simahawk committed:
edi.backend: improve exception handling

simahawk committed:
edi.backend: add docstrings and TODO

oca-travis committed:
[UPD] Update edi.pot

OCA-git-bot committed:
edi 13.0.1.5.0

simahawk committed:
edi: improve and fix tests

oca-travis committed:
[UPD] Update edi.pot

OCA-git-bot committed:
edi 13.0.1.6.0

simahawk committed:
edi: sort out components mixins, add input mixins

simahawk committed:
edi.backend: lookup component by direction too

simahawk committed:
edi: test improve fake components, add input base

simahawk committed:
edi: test process w/ fake components

etobella committed:
[IMP] edi: Refactoring views

etobella committed:
[IMP] edi: input processing, finishing tests

simahawk committed:
edi: fix consumer mixin tests

simahawk committed:
edi: add unique identifier

simahawk committed:
edi: add external identifier

simahawk committed:
edi: fix and organize menu items

simahawk committed:
edi: disable not ready cron

oca-travis committed:
[UPD] Update edi.pot

OCA-git-bot committed:
edi 13.0.1.7.0

simahawk committed:
edi.backend: return none when component not found

oca-travis committed:
[UPD] Update edi.pot

OCA-git-bot committed:
edi 13.0.1.8.0

etobella committed:
[IMP] edi: Security

simahawk committed:
edi: simplify edi events

Exclude the code from the name because it can be tricky to hook to
and overall it's a too specific event.

simahawk committed:
edi: simply ack handling

ACK is now provided by a specific type of exchange
and relies on the relation between records of given type.

simahawk committed:
edi.backend: simplify component lookup

etobella committed:
[FIX] edi: Missing filter on cron check

simahawk committed:
edi.backend: handle reception of files

oca-travis committed:
[UPD] Update edi.pot

OCA-git-bot committed:
edi 13.0.1.8.1

simahawk committed:
edi: move event triggers to exchange record

simahawk committed:
edi: move and rename _exchange_notify_record to exc rec

Notify the related record belongs now to the exchange record itself
as it makes more sense. It has been renamed for better meaning
to '_notify_related_record'.

simahawk committed:
edi.backend: rename check method for exchange_process

simahawk committed:
edi: rename test file for process

simahawk committed:
edi: cron tests silent logger

simahawk committed:
edi: cron generate ack record when needed

oca-travis committed:
[UPD] Update edi.pot

OCA-git-bot committed:
edi 13.0.1.9.0

oca-travis committed:
[UPD] Update edi.pot

OCA-git-bot committed:
edi 13.0.1.10.0

simahawk committed:
edi: add receive component base mixin

simahawk committed:
edi: add type advanced settings

simahawk committed:
edi: fix manifest description

simahawk committed:
edi: add base lookup for validators

simahawk committed:
edi.backend: fix handle reception of files

simahawk committed:
edi.backend fix _check_input_exchange_sync signature

oca-travis committed:
[UPD] Update edi.pot

OCA-git-bot committed:
edi 13.0.1.11.0

simahawk committed:
edi: improve validation handling

oca-travis committed:
[UPD] Update edi.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
edi 13.0.1.12.0

etobella committed:
[IMP] edi: Improve readme

etobella committed:
[IMP] edi: Allow to manually generate EDI Exchange records

OCA-git-bot committed:
edi 13.0.1.13.0

simahawk committed:
edi.backend: rename generate to match others

All the other main methods start with 'exchange_'.
Make generate respect this convention.

simahawk committed:
edi: add jobs configuration

etobella committed:
[FIX] edi: Rule not correctly created

OCA-git-bot committed:
edi 13.0.1.13.1

oca-travis committed:
[UPD] Update edi.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
edi 13.0.1.14.0

bosd committed:
Added translation using Weblate (Dutch)

bosd committed:
Translated using Weblate (Dutch)

Currently translated at 100.0% (2 of 2 strings)

Translation: edi-13.0/edi-13.0-base_edi
Translate-URL: https://translation.odoo-community.org/projects/edi-13-0/edi-13-0-base_edi/nl/

etobella committed:
[IMP] edi: Add a has_exchange_record function to search for the specific exchange record related

etobella committed:
[IMP] edi: Keep old field values on the fields_view_get

Without this, we are loosing the specific views of some fields

OCA-git-bot committed:
edi 13.0.1.14.1

simahawk committed:
edi: consumer mixin fix Serialized field usage

simahawk committed:
edi: component ease access to record

simahawk committed:
edi: refactor component lookup

The match is now done based on specific control keys: backend type, exchange type.
This makes components more clean and their behavior more understandable.

bealdav committed:
IMP purch..ubl: replace line_number by purch_line.id

OCA-git-bot committed:
purchase_order_ubl 13.0.1.2.0

OCA-git-bot committed:
edi 13.0.1.15.0

etobella committed:
[CHG] edi: Use more permissive licence: AGPL-> LGPL

etobella committed:
[CHG] edi: Use more permissive licence: AGPL-> LGPL

simahawk committed:
edi: cron call exchange action w/ jobs

simahawk committed:
edi: enable cron for input records

simahawk committed:
edi: add TODO for cron methods

simahawk committed:
edi: _get_component fallback to backend model

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
edi 13.0.1.15.1

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
base_edi 13.0.1.0.1

oca-travis committed:
[UPD] Update edi.pot

OCA-git-bot committed:
edi 13.0.1.16.0

acsonefho committed:
[FIX] edi: fix record property when there is no model to return the record's parent (if set) + unit test

rconjour committed:
[IMP] purchase_order_ubl: black, isort, prettier

rconjour committed:
[MIG] purchase_order_ubl: Migration to 14.0

rconjour committed:
[IMP] purchase_order_import: black, isort, prettier

rconjour committed:
[MIG] purchase_order_import: Migration to 14.0

PieterPaulussen commented on purchase_order_import/__manifest__.py:
```diff
@@ -3,14 +3,20 @@
+    "depends": [
+        "base_business_document_import",
+        "onchange_helper",
+        "purchase",
```
`"purchase"` is already in the dependency chain because of `"purchase_stock"`

rconjour replied:
You're right! Fxied

StefanRijnhart commented on purchase_order_import/security/ir.model.access.csv:
```diff
@@ -0,0 +1,3 @@
+id,name,model_id:id,group_id:id,perm_read,perm_write,perm_create,perm_unlink
+access_purchase_order_import,access_purchase_order_import,model_purchase_order_import,base.group_user,1,1,1,1
+access_order_response_import,access_order_response_import,model_order_response_import,base.group_user,1,1,1,1
```
Isn't the purchase user group more appropriate here?

StefanRijnhart commented on purchase_order_import/tests/test_order_response_import.py:
```diff
@@ -413,7 +413,8 @@ def test_13(self):
         self.assertEqual(self.purchase_order.state, "purchase")
-        self.assertEqual(len(self.purchase_order.picking_ids), 2)
+        self.assertEqual(len(self.purchase_order.picking_ids), 1)
+        self.assertTrue(self.purchase_order.picking_ids.backorder_id)
```
Wow isn't the picking backorder marked as a picking of the purchase order anymore? Do you know if this functional change is intentional in Odoo 14?

StefanRijnhart commented on purchase_order_import/tests/test_order_response_import.py:
```diff
@@ -426,11 +427,11 @@ def test_13(self):
         )
         move_backorder = move_ids.filtered(
-            lambda s: s.state == "assigned" and s.product_qty == 3
+            lambda s: s.state == "draft" and s.product_qty == 3
```
Here too, I am surprised that backorders in Odoo 14 are no longer confirmed automatically. Is this intentional on the side of Odoo?

StefanRijnhart replied:
Ah, I see that Odoo now daisy chains a number of wizards when transferring pickings. It always creates the backorder, and depending on how the user clicks their way across those wizards, the backorder pickings are either confirmed or cancelled. Maybe you can follow the picking confirmation flow in this test so that maybe you don't have to change what is actually tested.

etobella committed:
[FIX] edi: Fix security & filtering

etobella committed:
[FIX] edi_storage: sort key was not working properly

etobella committed:
[IMP] edi: Change security function

alexis-via committed:
base_business_document_import: Improve tests on taxes

Improve error handling
Code cleanup
Remove CREDITS: either we list all the companies that financially
supported the dev of the module or none, but we can't list just one.

OCA-git-bot committed:
edi 13.0.1.17.0

OCA-git-bot committed:
Merge PR #366 into 14.0

Signed-off-by alexis-via

oca-travis committed:
[UPD] Update base_business_document_import.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
base_business_document_import 14.0.2.0.0

OCA-git-bot committed:
[UPD] addons table in README.md

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: edi-14.0/edi-14.0-base_business_document_import
Translate-URL: https://translation.odoo-community.org/projects/edi-14-0/edi-14-0-base_business_document_import/

Yvesldff committed:
Added translation using Weblate (French (France))

Yvesldff committed:
Translated using Weblate (French (France))

Currently translated at 97.4% (38 of 39 strings)

Translation: edi-14.0/edi-14.0-base_business_document_import
Translate-URL: https://translation.odoo-community.org/projects/edi-14-0/edi-14-0-base_business_document_import/fr_FR/

bosd committed:
Added translation using Weblate (Dutch)

bosd committed:
Translated using Weblate (Dutch)

Currently translated at 56.4% (22 of 39 strings)

Translation: edi-14.0/edi-14.0-base_business_document_import
Translate-URL: https://translation.odoo-community.org/projects/edi-14-0/edi-14-0-base_business_document_import/nl/

bosd committed:
Translated using Weblate (Dutch)

Currently translated at 94.8% (37 of 39 strings)

Translation: edi-14.0/edi-14.0-base_business_document_import
Translate-URL: https://translation.odoo-community.org/projects/edi-14-0/edi-14-0-base_business_document_import/nl/

OCA-git-bot committed:
Merge PR #352 into 14.0

Signed-off-by StefanRijnhart

oca-travis committed:
[UPD] Update purchase_order_ubl.pot

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
[ADD] setup.py

OCA-git-bot committed:
[UPD] README.rst

etobella committed:
[MIG] base_edi: Migration to 14.0

etobella committed:
[IMP] edi_oca: black, isort, prettier

etobella committed:
[MIG] edi_oca: Migration to 14.0

OCA-git-bot committed:
Merge PR #370 into 14.0

Signed-off-by simahawk

oca-travis committed:
[UPD] Update base_edi.pot

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
[ADD] setup.py

OCA-git-bot committed:
[UPD] README.rst

phuctranfxvn committed:
[RM] Remove test-requirements.txt file

simahawk committed:
Merge pull request #374 from phuctranerp/14.0-remove-test-requirements

[14.0] Remove test-requirements.txt file (fix #373)

OCA-git-bot committed:
Merge PR #371 into 14.0

Signed-off-by etobella

oca-travis committed:
[UPD] Update edi_oca.pot

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
[ADD] setup.py

OCA-git-bot committed:
[UPD] README.rst

etobella committed:
[FIX] edi_oca: Missing changes of name

OCA-git-bot committed:
Merge PR #378 into 14.0

Signed-off-by simahawk

OCA-git-bot committed:
edi_oca 14.0.1.0.1

OCA-git-bot committed:
[UPD] addons table in README.md
